dm/pkg/shardddl/optimism/lock.go

```diff
@@ -790,11 +800,11 @@ func (l *Lock) trySyncForOneDDL(source, schema, table string, prevTable, postTab
 	}
 
 	tableCmp, tableErr := prevTable.Compare(postTable)
-	oldJoined, oldErr := l.joinNormalTables()
-
 	// Normal DDL
 	if tableErr == nil {
 		log.L().Debug("receive a normal DDL", zap.String("source", source), zap.String("schema", schema), zap.String("table", table), zap.Stringer("prevTable", prevTable), zap.Stringer("postTable", postTable))
+		oldJoined, oldErr := l.joinNormalTables()
```

Comment:
It looks like this could be put after L813 🤔

Reply:
No. L808-L809 update the prev_tables, we should calculate oldJoined before it.